### DISCUSSION TITLE
Shortcut for enable / disable scroll reverse

### DISCUSSION THREAD

**Discussion by EZeasy95** [pilotmoon/Scroll-Reverser#145] [Ideas]
Great app! One suggestion is to add keyboard shortcut that enables user to turn on / off scroll reverse!

**Comment by pilotmoon** [comment]
Thanks. There is an AppleScript interface for it, so you could srt up your own shortcut using Automator and a little script. It will be similar to how it works for PopClip: https://docs.pilotmoon.com/article/36-controlling-popclip-using-applescript

**Comment by marcuscps** [comment]
There is a way to do this without any proprietary apps/tools:

1 - Open the Automator app
2 - Create a "New Document"
3 - Select "Quick Action"
4 - Set "Workflow receives" to "no input"
5 - Drag the "Run AppleScript" item from the actions  library (left menu) and drop it into the right hand side panel.
6 - Paste the script below into the new script editor item:

```
tell application "Scroll Reverser"
	if enabled is true then
		tell application "Scroll Reverser" to set enabled to false
	else
		tell application "Scroll Reverser" to set enabled to true
	end if
end tell
```

7 - Save it as "Toggle Scroll Reverse enabled"
8 - Close the document (do NOT close Automator)
9 - Open the System Settings menu (from the Apple menu)
10 - Go to "Keyboard" -> "Keyboard Shortcuts ..."
11 - Select "Services" on the left menu
12 - Expand "General" on the right panel (if the item is not there, wait a bit a try again. Make sure Automator is running
13 - The "Toggle Scroll Reverse enabled" will be visible. Click the "none" twice and type your keyboard shortcut as desired
14 - Click done and close System Settings

Of course "Scroll Reverse" must be running, but it it is not, it will run the first time you use your shortcut.
You may see some warnings when running, but should not happen every time.

This should do the trick.
